client/src/common/client.ts

```diff
@@ -1340,6 +1348,13 @@ class DidSaveTextDocumentFeature extends DocumentNotifications<DidSaveTextDocume
 	}
 }
 
+class InvalidGlobError extends Error {
```

Comment:
I would leave this out and simply throw new Error. The client will not catch it anyways and the server will not care :-)

Reply:
Done! Though this does mean if another error occurs (that isn't converting the glob to a regex) inside the registration method, it may warn with a misleading message (and would also ignore the error). There's not a lot of code there to go wrong, but thought it worth mentioning.